hw/ip/otbn/util/shared/decode.py

```diff
@@ -31,13 +31,32 @@ def __init__(self, symbols: Dict[str, int], insns: Dict[int, int],
                 raise ValueError(
                     'No legal decoding for mnemonic: {}'.format(mnem))
             insn = INSNS_FILE.mnemonic_to_insn.get(mnem)
+            if insn is None:
+                raise ValueError(
+                    'No instruction found for mnemonic: {}'.format(mnem))
```

Comment:
This should never happen (because the mnemonics returned by `mnem_for_word` are always keys of `mnemonic_to_insn`). I'd probably just change line 33 to `INSNS_FILE.mnemonic_to_insn[mnem]`.

On a (slightly) related note, it's a pity that there's no support here for bogus words in the instruction stream. These don't matter as long as you don't execute them and I can imagine someone wanting to pad things with zeros or similar. Would it be hard to support? From the control flow perspective, they would behave the same as running off the top of memory. Thinking about it, I'm not sure that you require that `insns` gives a contiguous block of memory. Maybe you could just skip bogus words entirely, leaving gaps?

Reply:
Actually, if the bogus words are unreachable, they won't cause a problem. The graph is constructed by starting from either `__imem_start` or the subroutine start PC, and then traversing the possible paths; it doesn't parse every word in IMEM, even for the "whole program" version.

Reply:
Oh, I see! That makes sense.